### PR TITLE
[WIP] Dataset management

### PR DESCRIPTION
**Pull request by bnaul**
Separate data management from featurization; new "Data" tab handles uploading/transforming time series files. Featurization/prediction no longer handle data upload themselves.

Also adds a `TimeSeries` class that stores all the relevant {,meta}data about a single time series and handles saving/loading as netCDF4 files.

TODO:
1. Fix front-end tests
2. Documentation

### CODE REVIEW

File: mltsp/Flask/flask_app.py [hunk 1]
```diff
@@ -681,6 +659,86 @@ def list_featuresets(
         return authed_featuresets
 
 
+# TODO database module
+def list_datasets_cursor_to_html_table(cursor):
```
Comment (stefanv) [inline mltsp/Flask/flask_app.py]: Why do we operate directly on cursors?  Perhaps it is best to query the database, get the data into a format we can easily use, and then simply inject that data into the template.  Generating HTML in the library does not feel right.

File: mltsp/Flask/flask_app.py [hunk 2]
```diff
@@ -681,6 +659,86 @@ def list_featuresets(
...
+
+
+def list_datasets(
+        auth_only=True, by_project=False, name_only=False,
+        as_html_table_string=False):
+    """Return list of strings describing entries in 'datasets' table.
+
+    Parameters
+    ----------
+    auth_only : bool, optional
```
Comment (stefanv) [inline mltsp/Flask/flask_app.py]: All these options seem extraneous.  Do we ever need to query all the datasets in existence (i.e., can auth_only go)? Why not always return a dictionary of ``{'project': [dataset1, dataset2]}``, then ``by_project`` can go.  And why not always include the dates in the result?  And certainly never return as an HTML string.

File: mltsp/transformation.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import os
+import numpy as np
+from .cfg import config
+from mltsp import data_management
+from mltsp import time_series as tslib
+from sklearn import model_selection
+
+
+def transform_ts_files(input_paths, dataset_keys, transform_type):
```
Comment (stefanv) [inline mltsp/transformation.py]: We should move away from using file handles, and operate on TimeSeries where possible.

File: mltsp/time_series.py [hunk 4]
```diff
@@ -0,0 +1,161 @@
+import copy
+import netCDF4
+import numpy as np
+import pandas as pd
+import xarray as xr
+from mltsp.cfg import config
+
+
+def _default_values_like(old_values, value=None, upper=None):
```
Comment (stefanv) [inline mltsp/time_series.py]: What is this supposed to do?

File: mltsp/time_series.py [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+    else:
+        for new_array in new_values:
+            new_array[:] = np.linspace(lower, upper, len(new_array))
+
+    return new_values
+
+
+def make_array(x):
+    try:
+        x = np.asfarray(x).squeeze()
```
Comment (stefanv) [inline mltsp/time_series.py]: May introduce a copy--is that intended?  And why is the failure mode here to simply return the original?

File: mltsp/time_series.py [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+            if 'time' in ds:
+                t.append(ds.time.values)
+            if 'error' in ds:
+                e.append(ds.error.values)
+
+    return TimeSeries(make_array(t), make_array(m), make_array(e), target,
+                      meta_features, name, path)
+    
+
+class TimeSeries:
```
Comment (stefanv) [inline mltsp/time_series.py]: This class needs a long-ish docstring explaining the different data layouts.

File: mltsp/time_series.py [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+        self.error = e
+        self.target = target
+        self.meta_features = dict(meta_features)
+        self.name = name
+        self.path = path
+        self.channel_names = ["channel_{}".format(i)
+                              for i in range(self.n_channels)]
+    
+    # TODO name?
+    def channels(self):
```
Comment (stefanv) [inline mltsp/time_series.py]: Docstrings, docstrings, everywhere

File: mltsp/Flask/flask_app.py [hunk 8]
```diff
@@ -4,6 +4,7 @@
 from __future__ import print_function
 import sys
 import os
+from os.path import join as pjoin
 
 # list of (Google) admin accounts to have access to all projects
 sys_admin_emails = ['a.crellinquick@gmail.com']
```
Comment (profjsb) [inline mltsp/Flask/flask_app.py]: Why are we still hardcoding in Ari's email?

**Pull request merged**